### ISSUE TITLE
Build fails at `codesign` on OS X <10.13.6

### ISSUE THREAD

**Issue by malthejorgensen** [fish-shell/fish-shell#6792]
Building on OS X versions prior to 10.13.6 fails at the very end when running `codesign`.
The `-options runtime`-argument isn't available on these earlier versions of the OS.

### The error (output from `make`)

```
> make
[  0%] Built target CHECK-FISH-BUILD-VERSION-FILE
[  1%] Built target build_fish_pc
[ 21%] Built target pcre2-32
[ 91%] Built target fishlib
[ 92%] Linking CXX executable fish_key_reader
invalid argument "runtime"
make[3]: *** [fish_key_reader] Error 1
make[3]: *** Deleting file `fish_key_reader'
make[2]: *** [CMakeFiles/fish_key_reader.dir/all] Error 2
make[1]: *** [all] Error 2
make: *** [all] Error 2
```

**uname -a**

    Darwin google-ultron.lan 16.7.0 Darwin Kernel Version 16.7.0: Sun Jun  2 20:26:31 PDT 2019; root:xnu-3789.73.50~1/RELEASE_X86_64 x86_64

**sw_vers** (Mac OSX/macOS version)

    ProductName:	Mac OS X
    ProductVersion:	10.12.6
    BuildVersion:	16G2136

**fish --version** (after successful build –  PR https://github.com/fish-shell/fish-shell/pull/6791)

    fish, version 3.1.0-407-gc12ab3d29

**Comment by deepfriedfilth** [comment]
Was having issues installing via homebrew on OS X 10.9.5 (ancient, I know, but site states 10.6+) the other day as well, I'm guessing this was the underlying issue at hand.

Obviously you're trying to build/install from source, but have you tried installing via the [Installer/.pkg](https://github.com/fish-shell/fish-shell/releases/download/3.1.0/fish-3.1.0.pkg) option on their website?

Installation succeeds, but can only thing I can run without the following error is `fish -v`, just trying `fish` or `fish -h` results in:

```
dyld: lazy symbol binding failed: Symbol not found: ____chkstk_darwin
  Referenced from: /usr/local/bin/fish
  Expected in: /usr/lib/libSystem.B.dylib

dyld: Symbol not found: ____chkstk_darwin
  Referenced from: /usr/local/bin/fish
  Expected in: /usr/lib/libSystem.B.dylib

Trace/BPT trap: 5
```

**Comment by malthejorgensen** [comment]
@deepfriedfilth 

What you're getting is – unfortunately – a different issue. I just installed via the file you linked, and I'm getting the error you're describing. I'm fairly certain that that is unrelated to the issue described in this thread.

I think what you're getting is that the binary created simply isn't compatible with OS X 10.9, and I'm not sure it's possible create a binary that is compatible with both OS X 10.9 and 10.14, so the fish authors would have to provide two download links.

`homebrew` actually (theoretically at least) solves this by building the binary for you, when you do `brew install fish`.

However, currently you need to build from the latest `master` on git, since the bug https://github.com/fish-shell/fish-shell/pull/6688 is present in 3.1.0.
Aaaaand you need the fix in the PR referenced from this issue (PR https://github.com/fish-shell/fish-shell/pull/6791).

**Comment by malthejorgensen** [comment]
@deepfriedfilth 

Actually, I just looked at it some more. Your issue with the installer should be fixed in the next release. It was fixed with this commit https://github.com/fish-shell/fish-shell/commit/fbb0f79992d4bfa194e8f522d0d219a1dadd49f7, which went in one day after 3.1.0 was released.

This also means that 10.9 is the minimum supported OS X version (at least for the installer).

**Comment by deepfriedfilth** [comment]
Very informative. Thanks for the extra research and report!

😅 Also glad I made the support cut off, hope they release + update their site to reflect this soon.

**Comment by zanchey** [comment]
Should be fixed with #6791?

**Comment by malthejorgensen** [comment]
@zanchey You’re right was fixed by #6791. Closing.
